### PR TITLE
add user assitant on profile

### PR DESCRIPTION
**Pull request by stivo-m**
The purpose is to introduce a user assistant in the domain objects stored in the user profile. This is necessary when creating, and updating the user assistant.

### CODE REVIEW

File: lib/src/entities/user_profile.dart [hunk 1]
```diff
@@ -27,6 +27,10 @@ class UserProfile with _$UserProfile {
     @JsonKey(name: 'userBioData') BioData? userBioData,
     @JsonKey(name: 'homeAddress') Address? homeAddress,
     @JsonKey(name: 'workAddress') Address? workAddress,
+
+    /// [assistant] refers to the name of the Be.Well user assistant,
+    /// i.e Bev or Bowi.
+    @JsonKey(name: 'assistant') String? assistant,
```
Comment (mkdprime) [inline lib/src/entities/user_profile.dart]: An enum will be great here instead of string. With enum you can do validation with ease.

Comment (stivo-m) [reply lib/src/entities/user_profile.dart]: Okay. Looking into it.

File: lib/src/value_objects/enums.dart [hunk 2]
```diff
@@ -96,3 +96,23 @@ extension SupplierAccountTypeEx on SupplierAccountType {
...
+
+/// [AssistantExt] offers an extension to get string names of the [Assistant]
+extension AssistantExt on Assistant {
+  String get name {
+    switch (this) {
+      case Assistant.bev:
+        return 'BEV';
+      case Assistant.bowi:
+        return 'BOWI';
+      default:
```
Comment (vincentmichuki) [inline lib/src/value_objects/enums.dart:112]: here you can just have one `case` and the other as `default`.
Alternatively use an if statement
`if(this==Assistant.bev) {return 'BEV';} return 'BOWI';`

**Review by vincentmichuki — CHANGES_REQUESTED** [review]

File: lib/src/value_objects/enums.dart (hunk 2 above)
Comment (stivo-m) [reply lib/src/value_objects/enums.dart:112]: Okay. Fixing it.

File: lib/src/entities/user_profile.dart [hunk 3]
```diff
@@ -9,6 +9,15 @@ part 'user_profile.g.dart';
 /// [UserProfile] represents a valid user as mapped out from the backend
 /// In the event the backend mapping changes, this model should also change as well
 
+/// [_decodeAssistant] checks the String name of the assistant passed
+/// from the backend and returns a valid [Assistant]
+Assistant _decodeAssistant(dynamic value) {
```
Comment (mkdprime) [inline lib/src/entities/user_profile.dart]: The implementation and intent are not coherent.
ideally,`_decodeAssistant` expect a json string. But the implementation expects a normal string.
https://pub.dev/documentation/json_annotation/latest/json_annotation/JsonKey/fromJson.html

File: lib/src/entities/user_profile.dart [hunk 4]
```diff
@@ -27,6 +36,11 @@ class UserProfile with _$UserProfile {
     @JsonKey(name: 'userBioData') BioData? userBioData,
     @JsonKey(name: 'homeAddress') Address? homeAddress,
     @JsonKey(name: 'workAddress') Address? workAddress,
+
+    /// [assistant] refers to the name of the Be.Well user assistant,
+    /// i.e Bev or Bowi.
+    @JsonKey(name: 'assistant', fromJson: _decodeAssistant)
```
Comment (mkdprime) [inline lib/src/entities/user_profile.dart]: The `fromJson` part of the annotation seems like an overkil.
I expected a simple implementation like this -> https://github.com/savannahghi/domain_objects/blob/add-assistant/lib/src/entities/supplier_profile.dart#L28

Comment (stivo-m) [reply lib/src/entities/user_profile.dart]: Noted @daviddexter. I have utilized the alternative you suggested on this.

File: lib/src/entities/user_profile.dart (hunk 3 above)
Comment (stivo-m) [reply lib/src/entities/user_profile.dart]: This has been removed based on the discussion below.

File: lib/src/value_objects/enums.dart (hunk 2 above)
Comment (stivo-m) [reply lib/src/value_objects/enums.dart:112]: Fixed

File: lib/src/entities/user_profile.dart (hunk 1 above)
Comment (stivo-m) [reply lib/src/entities/user_profile.dart]: Fixed

**Review by daviddexter — APPROVED** [review]

**Review by vincentmichuki — APPROVED** [review]

**Pull request merged**